packages/core-bridge/src/helpers.rs

```diff
@@ -187,3 +187,56 @@ where
         cx.throw_type_error::<_, Vec<u8>>(format!("Invalid or missing {}", full_attr_path))
     }
 }
+
+// Recursively convert a Serde value to a JS value
+pub fn serde_value_to_js_value<'a>(cx: &mut impl Context<'a>, val: serde_json::Value) -> JsResult<'a, JsValue> {
+    match val {
+        serde_json::Value::String(s) => Ok(cx.string(s).upcast()),
+        serde_json::Value::Number(n) => Ok(cx.number(n.as_f64().unwrap()).upcast()),
+        serde_json::Value::Bool(b) => Ok(cx.boolean(b).upcast()),
+        serde_json::Value::Null => Ok(cx.null().upcast()),
+        serde_json::Value::Array(vec    ) => {
+            let arr: Handle<'a, JsArray> = JsArray::new(cx, vec.len() as u32);
+            for (i, v) in vec.into_iter().enumerate() {
+                let v = serde_value_to_js_value(cx, v)?;
+                arr.set(cx, i as u32, v)?;
+            }
+            Ok(arr.upcast())
+        }
+        serde_json::Value::Object(map) => {
+            let obj: Handle<'a, JsObject> = cx.empty_object();
+            for (k, v) in map {
+                let k = cx.string(snake_to_camel(k));
+                let v=  serde_value_to_js_value(cx, v)?;
+                obj.set(cx, k, v)?;
+            }
+            Ok(obj.upcast())
+        }
+    }
+}
+
+pub fn hashmap_to_js_value<'a>(cx: &mut impl Context<'a>, map: HashMap<String, serde_json::Value>) -> JsResult<'a, JsObject> {
+    let obj: Handle<'a, JsObject> = cx.empty_object();
+    for (k, v) in map {
+        let k = cx.string(snake_to_camel(k));
+        let v = serde_value_to_js_value(cx, v)?;
+        obj.set(cx, k, v)?;
+    }
+    Ok(obj)
+}
+
+fn snake_to_camel(s: String) -> String {
+    let mut result = String::new();
```

Comment:
nit: can probably optimize this but let's start with this and see if it becomes an issue (I predict it won't).

Reply:
Yeah, I doubt that will make a huge difference, but went for some obvious low hanging fruits, notably by not creating a new string unless there is actually a '_' somewhere, and by presizing the output string's buffer.Could also work at bytes level